x/fundraising/keeper/auction.go

```diff
@@ -221,3 +222,73 @@ func (k Keeper) CancelAuction(ctx sdk.Context, msg *types.MsgCancelAuction) (typ
 
 	return auction, nil
 }
+
+// AddAllowedBidders is a function for an external module and it simply adds new bidder(s) to AllowedBidder list.
+// Note that it doesn't do auctioneer verification because the module is generalized for broader use cases.
+// It is designed to delegate to an external module to add necessary verification and logics depending on their use case.
+func (k Keeper) AddAllowedBidders(ctx sdk.Context, auctionId uint64, bidders []types.AllowedBidder) error {
+	auction, found := k.GetAuction(ctx, auctionId)
+	if !found {
+		return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "auction %d is not found", auctionId)
+	}
+
+	if len(bidders) == 0 {
+		return types.ErrEmptyAllowedBidders
+	}
+
+	for _, bidder := range bidders {
+		if bidder.MaxBidAmount.IsNil() {
+			return types.ErrInvalidMaxBidAmount
+		}
+
+		if !bidder.MaxBidAmount.IsPositive() {
+			return types.ErrInvalidMaxBidAmount
+		}
+	}
+
+	// Add one or more bidders to allowed bidder list
+	if err := auction.SetAllowedBidders(bidders); err != nil {
```

Comment:
Does the method remove the existing allowed bidders?

Reply:
Great catch! I think it will remove the existing ones since there is no `append` from the existing allowed bidders. I will make the change and add test case for it. Thanks!

Reply:
@lubtd Can you review the fix when you have time?

Reply:
@lubtd Can you check the fix when you have time?